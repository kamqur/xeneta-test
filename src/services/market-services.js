import axios from "axios";

const getMarketRates = (origin, destination) =>
  axios.get(
    `https://685rp9jkj1.execute-api.eu-west-1.amazonaws.com/prod/rates?origin=${origin}&destination=${destination}`,
    {
      headers: {
        "x-api-key": process.env.REACT_APP_API_SECRET_KEY
      }
    }
  );

export { getMarketRates };
