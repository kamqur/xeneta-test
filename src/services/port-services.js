import axios from "axios";

const getPorts = () =>
  axios.get(
    `https://685rp9jkj1.execute-api.eu-west-1.amazonaws.com/prod/ports`,
    {
      headers: {
        "x-api-key": process.env.REACT_APP_API_SECRET_KEY
      }
    }
  );

export { getPorts };
