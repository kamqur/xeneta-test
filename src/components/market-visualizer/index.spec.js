import React from "react";
import { shallow, mount } from "enzyme";

import MarketVisualizer from "./index";
import Graph from "./graph";
import * as api from "../../services/market-services";

describe("MarketVisualizer Componenet", () => {
  it("snapshort testing", () => {
    const component = mount(
      <MarketVisualizer
        loadingMarketData={[]}
        marketError={[]}
        marketData={[]}
        currentDestination={""}
        currentOrigin={""}
        dispatch={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("marketdata is empty", () => {
    const component = shallow(
      <MarketVisualizer
        loadingMarketData={false}
        marketError={[]}
        marketData={[]}
        currentDestination={""}
        currentOrigin={""}
        dispatch={() => {}}
      />
    );

    const graph = component.find(Graph);
    expect(graph).toHaveLength(0);
  });

  it("marketdata is present", () => {
    const component = shallow(
      <MarketVisualizer
        loadingMarketData={false}
        marketError={[]}
        marketData={[1]}
        currentDestination={""}
        currentOrigin={""}
        dispatch={() => {}}
      />
    );

    const graph = component.find(Graph);
    expect(graph).toHaveLength(1);
  });

  it("currentOrigin/currentDestination is changed data fetch is called", () => {
    const mockGetData = jest
      .spyOn(api, "getMarketRates")
      .mockReturnValue(Promise.resolve([]));

    const component = mount(
      <MarketVisualizer
        loadingMarketData={false}
        marketError={[]}
        marketData={[]}
        currentDestination={""}
        currentOrigin={""}
        dispatch={() => {}}
      />
    );

    component.setProps({
      currentOrigin: { code: 1 },
      currentDestination: { code: 1 }
    });
  });

  it("currentOrigin/currentDestination is changed and data fails", () => {
    const mockGetData = jest
      .spyOn(api, "getMarketRates")
      .mockReturnValue(Promise.reject([]));

    const component = mount(
      <MarketVisualizer
        loadingMarketData={false}
        marketError={[]}
        marketData={[]}
        currentDestination={""}
        currentOrigin={""}
        dispatch={() => {}}
      />
    );

    component.setProps({
      currentOrigin: { code: 1 },
      currentDestination: { code: 1 }
    });
  });
});
