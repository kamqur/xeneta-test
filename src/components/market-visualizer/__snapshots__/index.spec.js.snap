// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketVisualizer Componenet snapshort testing 1`] = `
<MarketVisualizer
  currentDestination=""
  currentOrigin=""
  dispatch={[Function]}
  loadingMarketData={Array []}
  marketData={Array []}
  marketError={Array []}
>
  <div
    className="Market-visualizer-class-loading"
  >
    Loading market data
  </div>
</MarketVisualizer>
`;
